middleman_protocol/middleman_protocol/message.py

```diff
@@ -0,0 +1,387 @@
+from abc import ABC
+from abc import abstractmethod
+
+from mypy.types import Any
+
+from construct import Byte
+from construct import Bytes
+from construct import BytesInteger
+from construct import Container
+from construct import Enum
+from construct import GreedyBytes
+from construct import Int16ub
+from construct import PascalString
+from construct import StreamError
+from construct import Struct
+from construct import VarInt
+
+from golem_messages.cryptography import ecdsa_sign
+from golem_messages.cryptography import ecdsa_verify
+from golem_messages.exceptions import InvalidSignature
+from golem_messages.message.base import Message as BaseGolemMessage
+
+from .constants import ESCAPE_CHARACTER
+from .constants import ESCAPE_SEQUENCES
+from .constants import FRAME_PAYLOAD_LENGTH_BYTES_LENGTH
+from .constants import FRAME_PAYLOAD_LENGTH_STARTING_BYTE
+from .constants import FRAME_PAYLOAD_STARTING_BYTE
+from .constants import FRAME_REQUEST_ID_BYTES_LENGTH
+from .constants import FRAME_SEPARATOR
+from .constants import FRAME_SEPARATOR_LENGTH
+from .constants import FRAME_SIGNATURE_BYTES_LENGTH
+from .constants import PayloadType
+from .exceptions import FrameInvalidMiddlemanProtocolError
+from .exceptions import PayloadInvalidMiddlemanProtocolError
+from .exceptions import PayloadLengthMismatchMiddlemanProtocolError
+from .exceptions import PayloadTypeInvalidMiddlemanProtocolError
+from .exceptions import RequestIdInvalidTypeMiddlemanProtocolError
+from .exceptions import SignatureInvalidMiddlemanProtocolError
+from .registry import PAYLOAD_TYPE_TO_MIDDLEMAN_MESSAGE_CLASS
+from .registry import register
+
+
+class AbstractMiddlemanMessage(ABC):
```

Comment:
Why not `AbstractFrame`?

Reply:
I think it is more natural for `middleman protocol` to have `middleman messages`. But if you insist I can change it. But please consider also that the names of subclass will have to change accordingly.

Reply:
Of course, subclasses should change too.

If you really want it to be called `MiddlemanMessage` it's fine because it's not wrong but I would just prefer to avoid confusion with Golem messages. That's why I refer to them as frames in the blueprint. They're both frames of the binary protocol and messages in a high-level sense.

I just think that calling them frames makes things clearer. From the point of view of the Signing Service or Middleman it's more natural to think that the message is just the payload.

But it's just a suggestion. If you strongly prefer the current convention you can leave it as is.

Reply:
:+1: done